src/components/fields/textarea/textarea.tsx

```diff
@@ -96,15 +97,29 @@ export const Textarea = (props: IGenericFieldProps<ITextareaSchema>) => {
 		);
 	};
 
+	const renderLabel = () => {
+		const baseId = `${id}-base`;
+		if (typeof formattedLabel === "string") {
+			return <Form.Label htmlFor={baseId}>{formattedLabel}</Form.Label>;
+		} else {
+			return (
+				<Form.Label {...formattedLabel} htmlFor={baseId}>
+					{formattedLabel?.children}
+				</Form.Label>
+			);
+		}
+	};
+
 	return (
 		<>
-			<Form.CustomField label={formattedLabel} id={id} errorMessage={error?.message}>
+			<Form.CustomField id={id}>
+				{renderLabel()}
```

Comment:
what's the rationale for rendering label separately as well? would it be sufficient to remove the errorMessage prop but pass label to `Form.CustomField`?

Reply:
it was originally to make sure `htmlFor` points to the right id but i realized it might be unnecessary after all. test cases were also failing previously but they seem to work now, maybe i got misled by jest